datasets/goes/goes-glm/goes_glm.py

```diff
@@ -0,0 +1,71 @@
+import os
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import List, Union
+
+import pystac
+from stactools.goes_glm import stac
+from stactools.goes_glm.constants import DATACUBE_EXTENSION
+
+from pctasks.core.models.task import WaitTaskResult
+from pctasks.core.storage import StorageFactory
+from pctasks.dataset.collection import Collection
+
+GEOPARQUET_CONTAINER = "blob://goeseuwest/noaa-goes-geoparquet/"
+
+# The original GOES ETL code indicates duplicate assets. Is there a way to
+# handle duplicate assets in pctasks? -We should probably ingest items with
+# an "insert" rather than "upsert" action so it fails if duplicate source
+# data assets exist.
+
+
+class GoesGlmCollection(Collection):
+    @classmethod
+    def create_item(
+        cls, asset_uri: str, storage_factory: StorageFactory
+    ) -> Union[List[pystac.Item], WaitTaskResult]:
+        nc_storage, nc_asset_path = storage_factory.get_storage_for_file(asset_uri)
+
+        # download the netcdf
+        with TemporaryDirectory() as tmp:
+            tmp_dir = Path(tmp)
+            tmp_nc_asset_path = Path(tmp_dir, Path(nc_asset_path).name)
+            nc_storage.download_file(nc_asset_path, tmp_nc_asset_path)
+
+            # create item and geoparquet files (saved to same directory as the nc file)
+            item = stac.create_item(tmp_nc_asset_path)
+            tmp_parquets = {f.name: f.as_posix() for f in tmp_dir.glob("*.parquet")}
+
+            # preference: slim down the source netcdf asset
+            netcdf_asset_dict = item.assets["netcdf"].to_dict()
+            netcdf_asset_dict.pop("cube:dimensions")
+            netcdf_asset_dict.pop("cube:variables")
+            item.assets["netcdf"] = pystac.Asset.from_dict(netcdf_asset_dict)
+            item.stac_extensions.remove(DATACUBE_EXTENSION)
+
+            for suffix in ["events", "flashes", "groups"]:
+                # preference: remove the "geoparquet_" prefix from asset names
+                item.assets[suffix] = item.assets.pop(f"geoparquet_{suffix}")
+                # preference: remove "cloud-optimized" role to be consistent in the PC
+                item.assets[suffix].roles = ["data"]
```

Comment:
In https://github.com/stactools-packages/goes-glm/issues/23 we decided not to rename these

```suggestion
            for key in ["events", "flashes", "groups"]:
                # preference: remove "cloud-optimized" role to be consistent in the PC
                item.assets[key].roles = ["data"]
```

Reply:
No worries. I got a little ahead of things.

Reply:
Done.